components/kyma-environment-broker/internal/process/provisioning/ias_type.go

```diff
@@ -0,0 +1,69 @@
+package provisioning
+
+import (
+	"time"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal"
+	kebError "github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/error"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/ias"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	setIASTypeTimeout = 10 * time.Minute
+)
+
+type IASType struct {
+	bundleBuilder ias.BundleBuilder
+	disabled      bool
+}
+
+func NewIASType(builder ias.BundleBuilder, disabled bool) *IASType {
+	return &IASType{
+		bundleBuilder: builder,
+		disabled:      disabled,
+	}
+}
+
+func (s *IASType) Disabled() bool {
+	return s.disabled
+}
+
+func (s *IASType) ConfigureType(operation internal.ProvisioningOperation, runtimeURL string, log logrus.FieldLogger) (time.Duration, error) {
+	if s.disabled {
+		return 0, nil
+	}
+
+	spb := s.bundleBuilder.NewBundle(operation.InstanceID)
+	err := spb.FetchServiceProviderData()
+	if err != nil {
+		return s.handleError(operation, err, log, "fetching ServiceProvider data failed")
+	}
+
+	log.Infof("Configure SSO Type for ServiceProvider with RuntimeURL: %s", runtimeURL)
+	err = spb.ConfigureServiceProviderType(runtimeURL)
+	if err != nil {
+		return s.handleError(operation, err, log, "setting SSO Type failed")
+	}
+
+	return 0, nil
+}
+
+func (s *IASType) handleError(operation internal.ProvisioningOperation, err error, log logrus.FieldLogger, msg string) (time.Duration, error) {
+	log.Errorf("%s: %s", msg, err)
+	switch {
+	case kebError.IsTemporaryError(err):
+		if time.Since(operation.UpdatedAt) > setIASTypeTimeout {
```

Comment:
can we just use `return s.operationManager.RetryOperation(` here?

Reply:
it is because `RetryOperation` after reached timeout marks operation as failed, I decided to not mark the whole operation as failed just because this last sub-step failed (SKR works, dasbordURL is set) and only left an error log about failed this part of provisioning.

Reply:
if this is the way how it should work the :+1: